[BUG] download: HTML entities in canonical title are not unescaped (e.g. "&#x2F;" instead of "/")

### 🔄 Tested on Latest Release

- [x] I confirm that I can reproduce this issue on the latest version

### ✔️ Expected Behaviour

Downloaded ads should store the plain text title. A title containing "/" should be saved as "Foo / Bar".

### 🐞 Actual Behaviour

Since the canonical title is taken from the JSON metadata (published_ads_by_id), HTML entities are kept verbatim in the saved YAML and in the created folder name. Affected characters observed so far: "/" -> &#x2F;, "'" -> &#x27;, '"' -> &quot;

Two consequences:

1. The stored title is wrong. If such an ad is published again later, the entity text would go online.
2. The entity expands the string length, so the 65-character validation can fail for titles that are actually well within the limit. Example: a 57-character title containing two apostrophes becomes 67 characters once they are encoded as &#x27;, which aborts the download with "title length exceeds 65 characters".

In _resolve_download_title() the JSON title is only stripped, not unescaped:

    cached_ad = self.published_ads_by_id.get(ad_id)
    if cached_ad is not None:
        title = cached_ad.get("title")
        if isinstance(title, str) and title.strip():
            return title.strip()

The page-title fallback (_extract_title_from_ad_page) is not affected, since web_text already returns decoded text.

Possible fix: unescape the JSON title before returning it, e.g. html.unescape(title.strip()). Happy to open a PR if that direction is fine.

### 📋 Steps to Reproduce

Have an own ad whose title contains "/" (or an apostrophe), then run: kleinanzeigen-bot download --ads=all -- the saved YAML and the folder name contain "&#x2F;" instead of "/".

### 💻 What operating systems are you seeing the problem on?

Linux

### 📺 What browsers are you seeing the problem on? (if applicable)

Microsoft Edge

### 📃 Relevant log output (if applicable)

[INFO] Resolved title for ad 1234567890: [Foo &#x2F; Bar]
[INFO] Downloading ad to: /mnt/data/downloaded-ads/ad_1234567890_Foo &#x2F; Bar
[ERROR] - title: Value error, title length exceeds 65 characters

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/src/kleinanzeigen_bot/extract.py
+++ b/src/kleinanzeigen_bot/extract.py
@@ -3,6 +3,7 @@
 # SPDX-ArtifactOfProjectHomePage: https://github.com/Second-Hand-Friends/kleinanzeigen-bot/
 import asyncio
 import errno
+import html
 import os
 import stat
 import time
@@ -619,7 +620,7 @@ class AdExtractor(WebScrapingMixin):
         if cached_ad is not None:
             title = cached_ad.get("title")
             if isinstance(title, str) and title.strip():
-                return title.strip()
+                return html.unescape(title.strip())
 
         return await self._extract_title_from_ad_page()
 

--- a/tests/unit/test_extract.py
+++ b/tests/unit/test_extract.py
@@ -1011,6 +1011,17 @@ class TestAdExtractorContent:
         mock_extract_title.assert_not_called()
 
     @pytest.mark.asyncio
+    async def test_resolve_download_title_decodes_published_metadata_entities(self, test_extractor:extract_module.AdExtractor) -> None:
+        """Decode HTML entities only when using the cached manage-ads title."""
+        test_extractor.published_ads_by_id = {12345: {"id": 12345, "title": " Foo &#x2F; Bar &#x27;Test&#x27; &quot;Example&quot; "}}
+
+        with patch.object(test_extractor, "_extract_title_from_ad_page", new_callable = AsyncMock) as mock_extract_title:
+            title = await test_extractor._resolve_download_title(12345)
+
+        assert title == "Foo / Bar 'Test' \"Example\""
+        mock_extract_title.assert_not_called()
+
+    @pytest.mark.asyncio
     async def test_resolve_download_title_falls_back_to_page_title(self, test_extractor:extract_module.AdExtractor) -> None:
         """Keep manual page extraction unchanged when manage-ads metadata is missing."""
         test_extractor.published_ads_by_id = {12345: {"id": 12345, "title": "   "}}
@@ -1019,6 +1030,50 @@ class TestAdExtractorContent:
             title = await test_extractor._resolve_download_title(12345)
 
         assert title == "Page Title"
+
+    @pytest.mark.asyncio
+    async def test_cached_title_entities_are_decoded_before_title_validation(
+        self,
+        test_extractor:extract_module.AdExtractor,
+        tmp_path:Path,
+    ) -> None:
+        """Validate downloaded ad info with the decoded visible title, not encoded entity text."""
+        base_dir = tmp_path / "downloaded-ads"
+        base_dir.mkdir()
+        encoded_title = f"{'A' * 58} &#x2F; B"
+        decoded_title = f"{'A' * 58} / B"
+        test_extractor.published_ads_by_id = {12345: {"id": 12345, "title": encoded_title}}
+
+        page_mock = MagicMock()
+        page_mock.url = "https://www.kleinanzeigen.de/s-anzeige/test/12345"
+        test_extractor.page = page_mock
+
+        with (
+            patch.object(test_extractor, "web_text", new_callable = AsyncMock) as mock_web_text,
+            patch.multiple(
+                test_extractor,
+                web_execute = AsyncMock(return_value = {"universalAnalyticsOpts": {"dimensions": {"l3_category_id": "", "ad_attributes": ""}}}),
+                _extract_category_from_ad_page = AsyncMock(return_value = "160"),
+                _extract_special_attributes_from_ad_page = AsyncMock(return_value = {}),
+                _extract_pricing_info_from_ad_page = AsyncMock(return_value = (None, "NOT_APPLICABLE")),
+                _extract_shipping_info_from_ad_page = AsyncMock(return_value = ("NOT_APPLICABLE", None, None)),
+                _extract_sell_directly_from_ad_page = AsyncMock(return_value = False),
+                _download_images_from_ad_page = AsyncMock(return_value = []),
+                _extract_contact_from_ad_page = AsyncMock(return_value = ContactPartial()),
+            ),
+            patch.object(test_extractor, "_extract_title_from_ad_page", new_callable = AsyncMock) as mock_extract_title,
+        ):
+            mock_web_text.side_effect = ["Description text", "03.02.2025"]
+            ad_cfg, _staging_dir, final_dir, ad_file_stem = await test_extractor._extract_ad_page_info_with_directory_handling(base_dir, 12345)
+
+        assert len(encoded_title) > 65
+        assert len(decoded_title) <= 65
+        assert ad_cfg.title == decoded_title
+        assert "&#x2F;" not in ad_file_stem
+        assert "&#x2F;" not in final_dir.name
+        assert "/" not in ad_file_stem
+        assert "/" not in final_dir.name
+        mock_extract_title.assert_not_called()
 
     @pytest.mark.asyncio
     async def test_directory_handling_uses_published_title_for_names_and_ad_info(

